packages/next/next-server/server/render.tsx

```diff
@@ -521,6 +521,10 @@ export async function renderToHTML(
       previewData = tryGetPreviewData(req, res, previewProps)
     }
 
+    // Set environment variable flag for whether or not preview mode is active
+    if (!!previewData) setConfig()
+    process.env.__NEXT_PREVIEW_MODE = (!!previewData).toString()
```

Comment:
This will leak the variable to other requests causing the preview data to be set in places you don't expect it to. We'll have to figure out a better way to achieve this.